plugin/src/main/java/io/jenkins/plugins/analysis/warnings/Valgrind.java

```diff
@@ -0,0 +1,41 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.jenkinsci.Symbol;
+import hudson.Extension;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisModelParser;
+import io.jenkins.plugins.analysis.core.model.StaticAnalysisLabelProvider;
+import io.jenkins.plugins.analysis.core.model.SymbolIconLabelProvider;
+
+/**
+ * Provides a parser and customized messages for Android Lint.
+ *
+ * @author Michael Trimarchi
+ */
+public class Valgrind extends AnalysisModelParser {
+    private static final long serialVersionUID = 1L;
+    private static final String ID = "valgrind";
+
+    /** Creates a new instance of {@link Valgrind}. */
+    @DataBoundConstructor
+    public Valgrind() {
+        super();
+        // empty constructor required for stapler
+    }
+
+    /** Descriptor for this static analysis tool. */
+    @Symbol("valgrindParser")
```

Comment:
Is `valgrind` already used somewhere else?

Reply:
@uhafner I don't think is used at all. I add it because we are using in memory detection leak in iot devices.

Reply:
Sorry, I wasn't clear about this: wouldn't it be simpler to use the ID `valgrind` rather than `valgrindParser`?

Reply:
@uhafner I will repost with all you suggestions. Clean up and make a better commit. Including how use it and use cases

Reply:
@uhafner Done